src/cache/cache_las.c

```diff
@@ -426,16 +426,26 @@ __wt_las_page_skip_locked(WT_SESSION_IMPL *session, WT_REF *ref)
 	 * versions of data and all the updates are in the past.
 	 */
 	if (ref->page_las->skew_newest &&
-	    txn->read_timestamp > ref->page_las->unstable_timestamp)
+	    txn->read_timestamp > ref->page_las->unstable_durable_timestamp)
 		return (true);
 
 	/*
 	 * Skip lookaside pages if reading as of a timestamp, we evicted old
 	 * versions of data and all the unstable updates are in the future.
 	 */
-	if (!ref->page_las->skew_newest &&
-	    txn->read_timestamp < ref->page_las->unstable_timestamp)
-		return (true);
+	if (!ref->page_las->skew_newest) {
+		/*
+		 * Skip lookaside pages during checkpoint if all the unstable
+		 * durable updates are in the future.
+		 */
+		if (WT_SESSION_IS_CHECKPOINT(session) &&
+		    txn->read_timestamp <
+		    ref->page_las->unstable_durable_timestamp)
+			return (true);
+
+		if (txn->read_timestamp < ref->page_las->unstable_timestamp)
+			return (true);
```

Comment:
This is the only code that uses `unstable_timestamp` anymore, why is it important for the two cases to be dealt with differently?

i.e: what is the advantage of tracking both `unstable_timestamp` and `unstable_durable_timestamp`?

Reply:
`unstable_timestamp` is required for ordinary cursor reads. An ordinary read should not skip reading LAS if `read_timestamp(210)` < `durable_timestamp(220)` but `read_timestamp(210)` > `commit_timestamp(200)`.  Ordinary read should still honor the `commit_timestamp` stored in the `unstable_timestamp`. That's why we still track the `unstable_timestamp`.

Reply:
Does that mean that keeping just the unstable check here would be correct, but might mean that checkpoints read more content from lookaside than necessary?

Reply:
That's right.

Reply:
It would be valuable to capture that in a comment somewhere (probably here). Could you expand the comments?